src/clj/rems/db/category.clj

```diff
@@ -0,0 +1,82 @@
+(ns rems.db.category
+  (:require [clojure.tools.logging :as log]
+            [rems.db.core :as db]
+            [rems.json :as json]
+            [rems.schema-base :as schema-base]
+            [rems.common.util :refer [build-index]]
+            [schema.coerce :as coerce]
+            [schema.core :as s]
+            [medley.core :refer [assoc-some]]))
+
+(s/defschema CategoryData
+  {(s/optional-key :description) schema-base/LocalizedString
+   (s/optional-key :children) [{:id s/Int}]})
+
+(def ^:private validate-categorydata
+  (s/validator CategoryData))
+
+(s/defschema CategoryDb
+  (-> {:id s/Int
+       :organization schema-base/OrganizationId}
+      (merge CategoryData)))
+
+(def ^:private coerce-CategoryDb
+  (coerce/coercer! CategoryDb coerce/string-coercion-matcher))
+
+(defn- format-category [category]
+  (let [categorydata (json/parse-string (:categorydata category))]
+    (-> category
+        (update :organization (fn [organization-id] {:organization/id organization-id}))
+        (dissoc :categorydata)
+        (merge categorydata))))
+
+(def ^:private categories-cache (atom nil))
+
+(defn reload-cache! []
+  (log/info :start #'reload-cache!)
+  (reset! categories-cache
+          (build-index {:keys [:id] :value-fn identity}
+                       (map (comp coerce-CategoryDb format-category)
+                            (db/get-categories))))
+  (log/info :end #'reload-cache!))
+
+(defn get-category
+  "Get a single category by id"
+  [id]
+  (when (nil? @categories-cache)
+    (reload-cache!))
```

Comment:
In the DUO/Mondo features I tried to avoid this sort of code by having that one internal endpoint that does the lazy load. We can keep it like this and see if we come up with other needs in the future.

Reply:
yeah true! we can extract the loading if the cache needs to be warmed like this elsewhere too.